### DISCUSSION TITLE
Get corresponding PR from Github deployment status webhook

### DISCUSSION THREAD

**Discussion by joostmeijles** [octokit/octokit.js#2016] [Q&A]
I am creating a Github app that listens for deployment_status.created webhook events. The webhook is triggered when a PR is automatically deployed (using a 3rd party Github app).

When the deployment has a certain state, e.g. success, I want to add a comment to the corresponding PR.

My example code:
```
app.on('deployment_status.created', async (context) => {
    const deployment_status = context.payload.deployment_status

    if (deployment_status.state === 'success') {
      // TODO: comment on the corresponding PR
    }
  })
```
How do I get the corresponding PR?

NB. There seems to be nothing in the payload for it (https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#deployment_status)

**Comment by gr2m** [comment]
I _think_ that `context.payload.deployment.sha` is the commit sha. With the commit sha, you can retrieve the pull request(s) that are associated with that commit: https://docs.github.com/en/rest/reference/repos#list-pull-requests-associated-with-a-commit

**Reply by gr2m** [reply]
A good place to ask GitHub API questions is https://github.community/c/github-ecosystem/37. There are several hubbers monitoring this forum, they will be better able to help you. I'm just maintaining the JavaScript Octkit SDKs and libraries

**Reply by joostmeijles** [reply]
Asked in the ecosystem: https://github.community/t/get-corresponding-pr-from-github-deployment-status-webhook/162668

Thanks!
